Translations: support for plurals (#2837)
slint-ui/slint: docs/language/src/concepts/translations.md

## Plurals

> A special kind of formatting is the plural formating, because the string may change if there is a single elemnts or several.

Reviewer: ```suggestion
A special kind of formatting is the plural formatting, because the string may change depending on if there is a single element or several.
```

> So given `count`, an expression that represent the count of something, we can form the plural with the `|` and `%` symbols like so:
> `@tr("I have {n} item" | "I have {n} items" % count)`.
> Use `{n}` in the format string to access the expression after the `%`

Reviewer: ```suggestion
Given `count` and an expression that represents the count of something, we can form the plural with the `|` and `%` symbols like so:
`@tr("I have {n} item" | "I have {n} items" % count)`.
Use `{n}` in the format string to access the expression after the `%`.
```

## Extracting the String from the files

> We provide a `slint-tr-extractor` tool that parse .slint files and can generate a .po file.

Reviewer: ```suggestion
Use the `slint-tr-extractor` tool to generate a .po file from .slint files.
```

> You can run it like so

Reviewer: ```suggestion
You can run it like so:
```

```sh
find -name \*.slint | xargs slint-tr-extractor -o MY_PROJECT.pot
```

> This will create a file called `MY_PROJECT.pot`
> (but you should replace MY_PROJECT with your actual project name, see bellow for naming)

Reviewer: ```suggestion
This will create a file called `MY_PROJECT.pot`. Replace MY_PROJECT with your actual project name. To learn how the project name affects the lookup of translations, see the sections below.
```

> ## Doing the translations

Reviewer: ```suggestion
## Translating Your Application
```

> `.pot` file are [Gettext](https://www.gnu.org/software/gettext/) template files. It is the same as a `.po` file, but doesn't contain actual
> translations. They can be edited by hand with a text editor, or there are a few tools you can use to translate them, option includes
>  - [poedit](https://poedit.net/)
>  - [OmegaT](https://omegat.org/)
>  - [Lokalize](https://userbase.kde.org/Lokalize)
>  - [Transifex](https://www.transifex.com/) (web interface)

Reviewer: I think we should have an introduction at the top of the file that outlines _all_ of the steps, i.e.

1. Marking translatable strings with @tr.
2. Extracting the tr marked strings into .pot files.
3. Generating translations by using various editor choices to create .po files from .pot files.
4. Compiling .po files to .mo files.
5. Loading them at run-time with Slint APIs.

Right now all these steps are scattered. Some here, some further down. I think it makes sense to outline all the steps first, to provide the big picture, and then go into details in the sections.

```suggestion
`.pot` file are [Gettext](https://www.gnu.org/software/gettext/) template files. It is the same as a `.po` file, but doesn't contain actual
translations. They can be edited by hand with a text editor, or there are a few tools you can use to translate them, option include:
 - [poedit](https://poedit.net/)
 - [OmegaT](https://omegat.org/)
 - [Lokalize](https://userbase.kde.org/Lokalize)
 - [Transifex](https://www.transifex.com/) (web interface)
```

Author: > Right now all these steps are scattered. Some here, some further down. I think it makes sense to outline all the steps first, to provide the big picture, and then go into details in the sections.

What your describe is just the outline of the document.  (But maybe the syntax of @tr can be made sub-sections instead of sections)

## Loading the translations at runtime

> [Gettext](https://www.gnu.org/software/gettext/) is used at runtime to get the translations.

Reviewer: "get" is a bit unspecific here. I suggest to remove this sentence in favor of the aforementioned translation.

So the first thing to do is to convert the `.po` files in `.mo` files that the gettext runtime can open.

> This can be done with the `msgfmt` from the gettext provided tools.

Reviewer: ```suggestion
This can be done with the `msgfmt` command line tool from the gettext package.
```
